File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/AspPageBox/Functionality/InteractWithASPNETPage.vb
Namespace CompanionKit.Controls.AspPageBox.Functionality

	Public Class InteractWithASPNETPage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

	        

		End Sub

        Private Sub InteractWithASPNETPage_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Me.demoAspPageBox.Path = "Controls\AspPageBox\Functionality\AspPageFormVB.aspx"
        End Sub

        <Serializable()> _
        Public Class MyAspPageBox
            Inherits Gizmox.WebGUI.Forms.Hosts.AspPageBox
            ' Methods
            Protected Overrides Sub FireEvent(ByVal objEvent As IEvent)
                MyBase.FireEvent(objEvent)
                If (objEvent.Type = "MessageBox") Then
                    MessageBox.Show(objEvent.Item("message"))
                End If
            End Sub

        End Class

    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/AspPageBox/Functionality/InteractWithASPNETPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.AspPageBox.Functionality
{
    public partial class InteractWithASPNETPage : UserControl
    {
        public InteractWithASPNETPage()
        {
            InitializeComponent();
        }

        private void InteractWithASPNETPage_Load(object sender, EventArgs e)
        {
            this.demoAspPageBox.Path = @"Controls\AspPageBox\Functionality\asppageform.aspx";
        }

        [Serializable()]
        public class MyAspPageBox : Gizmox.WebGUI.Forms.Hosts.AspPageBox
        {

            protected override void FireEvent(Gizmox.WebGUI.Common.Interfaces.IEvent objEvent)
            {
                base.FireEvent(objEvent);

                if (objEvent.Type == "MessageBox")
                {
                    MessageBox.Show(objEvent["message"]);
                }
            }
        }
    }
}